datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -86,45 +86,54 @@ func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig)
 	var (
 		repo         Repository
 		checkpointDB CheckpointDB
-		setter       ComponentSetter
+		setter       []ComponentSetter
 		err          error
 	)
 	switch cfg.Mode {
 	case ModeKAS:
-		newRepo, err := kas.NewRepository(cfg.KasConfig)
+		repo, checkpointDB, setter, err = getKasComponents(cfg.KasConfig)
 		if err != nil {
-			logger.Error("Failed to create new Repository", "err", err, "user", cfg.KasConfig.DBUser, "host", cfg.KasConfig.DBHost, "port", cfg.KasConfig.DBPort, "name", cfg.KasConfig.DBName, "cacheUrl", cfg.KasConfig.CacheInvalidationURL, "x-chain-id", cfg.KasConfig.XChainId)
```

Comment:
Why did you erased the log?

Reply:
This is a duplicate log. In the `NewRepository` methods for both KAS and Kafka, there is an error log when the creation is failed.